_posts/2023-03-30-technical-roadmap-opensearch-extensibility.markdown

```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title:  "Technical roadmap: OpenSearch extensibility"
+authors:
+  - dblock
+date:   2023-03-30
+categories:
+  - technical-post
+meta_keywords: OpenSearch extensibility, extensions framework, OpenSearch plugins, OpenSearch Dashboards
+meta_description: Learn how the OpenSearch Project is replacing its plugin framework with a new catalog of extensions to help users and developers overcome extensibility challenges.
+excerpt: The OpenSearch plugin framework presents a number of challenges for users and developers. To solve these, we've embarked on a journey to replace the plugin mechanism with extensions.
+---
+
+The primary reason users choose OpenSearch is the wide range of use cases they can address with its features, such as search or log analytics. Thus, we aim to make the OpenSearch Project the preferred platform for builders by creating a vibrant and deeply integrated ecosystem of projects, features, content packs, integrations, and tools that can be found quickly, installed securely, combined to solve problems, and monetized by many participants. 
+
+The existing mechanism used to extend OpenSearch and OpenSearch Dashboards is a plugin framework. It provides a useful way to extend functionality, particularly when the new functionality needs access to a significant number of internal APIs. However, the plugin framework presents a number of challenges for users and developers in the areas of administration, dependency management, security, availability, scalability, and developer velocity. To begin solving these, we've embarked on a journey to replace the OpenSearch plugin mechanism with a new catalog of _extensions_. We plan to ship two new SDKs for OpenSearch and OpenSearch Dashboards, and then launch a catalog of extensions.
+
+In this blog post, we'll introduce the concept of extensions and outline some proposed projects in this area.
```

Comment:
There's a lot of talk below about what we "will" do. Could we add a simple disclaimer to the end of this sentence saying something like, "This blog post reflects some of our current project plans, but we want your feedback! Our plans are subject to change based on feedback, contributions, and available resources." Trying not to make it look like less of a stale legal disclaimer.

Reply:
I said "This blog post reflects some of our current project plans. Like all plans, they may change as we make progress, and we would love your help!"